File: korrida/day_17/4.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> divs(int n)
{
    vector<int> d;
    for (int i = 2; i < n / +1; ++i)
    {
        if (n % i == 0)
        {
            d.push_back(i);
        }
    }

    return d;
}

int main()
{
    for (int n = 50_000_000, n < 60_000_000; ++n)
    {
    }
    return 0;
}

File: korrida/day_17/4.py
def divs(number):
    return [i for i in range(2, number // 2 + 1) if number % i == 0]


c = 0
minn = 100 ** 10
for n in range(50_000_000, 60_000_000 + 1):
    if len(divs(n)) == 6 and 911 in divs(n):
        c += 1
        minn = min(minn, n)